### PR TITLE
72 export metadata

### PR DESCRIPTION
**Pull request by chStaiger**
- Extended meta.py with function that exports some system metadata and user metadata as python dictionary
- Added a function that iterates over a whole collection and exports all metadata of the collection, its subcollections and all data objects as python dictionary.

Dictionary structure:
```
{
'name': name, 
'irods_id': id,
'metadata': [(avu1), (avu2), ...],
'subcollections':[
    {'name': name, 
     'irods_id': id,
      'rel_path': relative path to the parent collection #recursive export,
      'metadata': [(avu1), (avu2), ...]
    }, 
    ...
]
'dataobjects':[{
    'name': name, 
    'irods_id': id,
    'rel_path': relative path to the parent collection #recursive export,
    'checksum': object checksum
    'metadata': [(avu1), (avu2), ...]
   },
  ...
  ]
}
```

### CODE REVIEW

**Comment by chStaiger** [comment]
- the to_json function in metadata are added to the intgeration tests, the format of the json seems to be ok for now
- the export function is still a prototype and we will have to figure out how it should work during the coming projects. The use case is not entirley clear yet.

@maartenschermer and @qubixes do you agree?

If yes, please approve this PR.

**Review by qubixes — APPROVED** [review]
Approved!

File: ibridges/meta.py [hunk 1]
```diff
@@ -142,3 +144,31 @@ def clear(self):
...
+        keys:
+            List of Attribute names which should be exported to "metadata". 
+            By default all will be exported.
+
+        """
+        meta_dict: dict[str, Any] = {}
+        meta_dict["name"] = self.item.name
+        meta_dict["irods_id"] = self.item.id
+        if is_dataobject(self.item):
+            meta_dict["checksum"] = self.item.checksum
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/meta.py:169]: I would separate that from the export metadata. We can think of a function that iterates over all objects in a collection and enforces the echecksum.
The reason for separating the two functionalities is mainly running time. If you uploaded a large collection without checksums and then want to export the metadata, the command can hang for a long time if first all checksums have to be calculated.

File: ibridges/export_metadata.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            …
+        ]
+    }
+    """
+    if is_dataobject(meta.item):
+        return meta.to_dict(keys = keys)
+    if is_collection(meta.item):
+        metadata_dict = meta.to_dict(keys = keys)
+        if recursive is True:
+            metadata_dict["ibridges_metadata_version"] = "1.0"
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/export_metadata.py:57]: Yep, let's see how tp get that in there ;)

[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridges/export_metadata.py]: Done

File: ibridges/meta.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maartenschermer) [reply ibridges/meta.py:169]: Ok, that makes sense.

**Comment by maartenschermer** [comment]
Approved!

**Pull request merged**